### PR TITLE
JDK-8320859: gtest high malloc footprint caused by BufferNodeAllocator stress test

### PR DESCRIPTION
**Pull request by tstuefe**
`BufferNodeAllocatorTest.stress_free_list_allocator_vm` is too expensive. On my box, this test accumulates ~1.5 GB of malloc footprint and raises the libc memory retention from 50m to 800m.

It is quite alone in its hunger. The rest of the tests together accumulate just ~50mb of libc retention.

The buffer does a stress test of the BufferNodeAllocator. Four "mutator threads" race four "gc threads". Mutator threads allocate buffers, GC threads release them. No processing is done on the buffers; this seems to be purely a test of the allocator and its freelist mechanism. The memory footprint of this test depends on the number of retained free buffers in the allocator. The allocator bulk-releases free buffers triggered by a free-count threshold. On my box, the number of free items in the unmodified stock VM is 100k..200k.

It is not clear whether the fact that so many free list items exist indicates a problem with the allocator itself. It looks like the free list should be drained if there are more than 10 items in this list.

In any case, each buffer's capacity is 1024 * sizeof(void*)+header, so 8KB+x, and with 100..200k of those things it explains the NMT-reported malloc footprint of ~1..2GB. An easy fix is to reduce the size of this buffer; since they are not processed, their size should not matter.

---------

The patch decreases the size per buffer node to cache line size. I chose cache line size out of the vague feeling that I don't want to cause false sharing and thereby degrade the test. 

Interestingly enough, reducing the buffer size greatly increases the number of free items since it makes allocation cheaper. This also happens in release builds, so this is not us zapping stuff; my unproven assumption is that this is the libc just being slower when allocating 8K vs 64-byte blocks. But with smaller buffers, we spend more time in freelist management and less time in the libc, which is good for a stress test.

This patch also increases the number of "processor" threads vs "mutator" threads. Since allocation seems to have a speed edge over deallocation, this reduces the number of free items somewhat.

This reduces the malloc spike seen from this thread from 1.5-2GB to ~160MB (release) 3xxMB (debug). We could reduce it a lot more if we reduced the buffer size to its minimum (1 slot); that would risk false sharing and may degrade test performance.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8320859](https://bugs.openjdk.org/browse/JDK-8320859): gtest high malloc footprint caused by BufferNodeAllocator stress test (**Enhancement** - P4)

### Reviewers
 * [Albert Mingkun Yang](https://openjdk.org/census#ayang) (@albertnetymk - **Reviewer**) ⚠️ Review applies to [7a8a694f](https://git.openjdk.org/jdk/pull/16845/files/7a8a694faf48c02c8c08b674b61d08674652c69a)
 * [Kim Barrett](https://openjdk.org/census#kbarrett) (@kimbarrett - **Reviewer**) ⚠️ Review applies to [7a8a694f](https://git.openjdk.org/jdk/pull/16845/files/7a8a694faf48c02c8c08b674b61d08674652c69a)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/16845/head:pull/16845` \
`$ git checkout pull/16845`

Update a local copy of the PR: \
`$ git checkout pull/16845` \
`$ git pull https://git.openjdk.org/jdk.git pull/16845/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 16845`

View PR using the GUI difftool: \
`$ git pr show -t 16845`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/16845.diff">https://git.openjdk.org/jdk/pull/16845.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/16845#issuecomment-1829627754)

### CODE REVIEW

File: test/hotspot/gtest/gc/shared/test_bufferNodeAllocator.cpp [hunk 1]
```diff
@@ -185,25 +185,30 @@ class BufferNode::TestSupport::ProcessorThread : public JavaTestThread {
 };
 
 static void run_test(BufferNode::Allocator* allocator, CompletedList* cbl) {
-  const uint nthreads = 4;
+
+  // deallocation is slower than allocation, so lets create
+  // more deallocation threads to prevent too large buildup of
+  // free nodes (footprint)
+  const uint nthreads_mut = 4;
+  const uint nthreads_proc = 6;
```
Comment (albertnetymk) [inline test/hotspot/gtest/gc/shared/test_bufferNodeAllocator.cpp:193]: Naming them "num_processor_threads" and "num_allocator_threads" is more consistent with the corresponding types `ProcessorThread` and `AllocatorThread`, IMO.

**Review by albertnetymk — APPROVED** [review]

**Review by kimbarrett — APPROVED** [review]
Looks good.

**Comment by tstuefe** [comment]
Thanks @kimbarrett and @albertnetymk. I will wait the obligatory 24 hrs before pushing tomorrow.

**Pull request closed without merging**